iquip/monitor.py

```diff
@@ -0,0 +1,116 @@
+"""Module for monitoring devices."""
+
+from typing import Any, TypeVar, Generic, Optional, Callable
+
+from PyQt5.QtWidgets import QWidget, QLabel, QHBoxLayout
+from PyQt5.QtCore import pyqtSignal
+
+T = TypeVar("T")
+
+
+class Monitor(Generic[T]):
+    """A basic monitor class with a generic value type T.
+    
+    Monitors watch a single value, and notify any change of the value by calling
+    a callback function.
+
+    This class can be used as a concrete Monitor if the usage is simple enough
+    and there is no need to implement self._read().
+    """
+
+    def __init__(
+        self,
+        initial_value: T,
+        updated_callback: Optional[Callable[[T], Any]] = None,
+    ):
+        """
+        Args:
```

Comment:
Please add `Extended` into the docstring!

Reply:
Hmm, actually this is not _extended_ because:
1. I did not use `super().__init__()`, so it would be _overridden_ instead.
2. However, it is not overridden, since `Generic` does not define `__init__()`.

Reply:
Oh, I see! Then, there is no problem.